kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -402,41 +489,56 @@ define([
          */
         function render() {
             return Promise.try(() => {
-                const events = Events.make(),
-                    inputControl = makeInputControl(),
+                const inputControl = makeInputControl(),
                     content = div({ class: 'input-group', style: { width: '100%' } }, inputControl);
 
                 ui.setContent('input-container', content);
-
-                $(ui.getElement('input-container.input'))
-                    .select2({
-                        allowClear: true,
-                        templateResult: formatObjectDisplay,
-                        templateSelection: selectionTemplate,
-                        ajax: {
-                            delay: 250,
-                            transport: function (params, success, failure) {
-                                return fetchData(params.data.term)
-                                    .then((data) => {
-                                        success({ results: data });
-                                    })
-                                    .catch((err) => {
-                                        console.error(err);
-                                        failure(err);
-                                    });
-                            },
+                const dropdown = $(ui.getElement('input-container.input'));
+                dropdown.select2({
+                    allowClear: true,
+                    templateResult: formatObjectDisplay,
+                    templateSelection: selectionTemplate,
+                    ajax: {
+                        delay: 250,
+                        processResults: (data) => {
+                            // update the currently selected one if applicable
+                            const currentData = dropdown.select2('data')[0];
```

Comment:
This fixes the issue where display data doesn't get updated. So if you start with `undefined` stuff in the currently selected option, then do a search that brings in a matching value, this will update that option with data from the service.